Expose dist through make help

sharidas opened the pull request:
Expose dist through make help.
This would help users to build the app and use it.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

tomneedham commented:
I would make `make` do the composer install, and `make dist` do the packaging - like most other apps.

sharidas commented:
Current state of the PR:
`make help` would provide users:

```
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile  make help
help                    Show this help message
dist                    Build distribution and create tar file impersonate.tar.gz
build-dep               fetch all dependencies
js-templates            build templates for frontend
clean                   Clean
build-src               Build source package
---------------------
 Tests
---------------------
test-php-unit           Run php unit tests
test-php-unit-dbg       Run php unit tests using phpdbg
test-php-style          Run php-cs-fixer and check owncloud code-style
test-php-style-fix      Run php-cs-fixer and fix code style issues
test-php-phan           Run phan
test-php-phpstan        Run phpstan
test-acceptance-api     Run API acceptance tests
test-acceptance-cli     Run CLI acceptance tests
test-acceptance-webui   Run webUI acceptance tests
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile  
```

Below are the outputs of the make 
1.  clean
```
sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile  make clean
rm -fR build
rm -f js/templates/impersonateNotification.handlebars.js js/templates/addImpersonateIcon.handlebars.js js/templates/removeImpersonateIcon.handlebars.js
rm -fr /home/sujith/test/owncloud3/apps/impersonate/node_modules
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile 
```

2. dist
```
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile  make dist
/usr/bin/npm install /home/sujith/test/owncloud3/apps/impersonate && touch node_modules
added 9 packages from 43 contributors and audited 10 packages in 0.907s
found 0 vulnerabilities



   ╭───────────────────────────────────────────────────────────────╮
   │                                                               │
   │      New minor version of npm available! 6.4.1 -> 6.7.0       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v6.7.0   │
   │               Run npm install -g npm to update!               │
   │                                                               │
   ╰───────────────────────────────────────────────────────────────╯

/home/sujith/test/owncloud3/apps/impersonate/node_modules/handlebars/bin/handlebars -n "OCA.Impersonate" js/templates/impersonateNotification.handlebars -f js/templates/impersonateNotification.handlebars.js
/home/sujith/test/owncloud3/apps/impersonate/node_modules/handlebars/bin/handlebars -n "OCA.Impersonate" js/templates/addImpersonateIcon.handlebars -f js/templates/addImpersonateIcon.handlebars.js
/home/sujith/test/owncloud3/apps/impersonate/node_modules/handlebars/bin/handlebars -n "OCA.Impersonate" js/templates/removeImpersonateIcon.handlebars -f js/templates/removeImpersonateIcon.handlebars.js
mkdir -p build/impersonate
cp --parents -r \
        appinfo \
        controller \
        css \
        img \
        js \
        l10n \
        lib \
        README.md \
        LICENSE \
        CHANGELOG.md \
        templates \
        build/impersonate
rm -f build/impersonate/js/templates/*.handlebars
Skipping signing, either no key and certificate found in /home/sujith/.owncloud/certificates/impersonate.key and /home/sujith/.owncloud/certificates/impersonate.crt or occ can not be found at /home/sujith/test/owncloud3/apps/impersonate/../../occ
tar -czf build/impersonate.tar.gz -C /home/sujith/test/owncloud3/apps/impersonate/build/impersonate ../impersonate
tar: Removing leading `../' from member names
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile 
```
3. build-dep
```
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile  make build-dep
/usr/bin/npm install /home/sujith/test/owncloud3/apps/impersonate && touch node_modules
added 9 packages from 43 contributors and audited 10 packages in 1.741s
found 0 vulnerabilities



   ╭───────────────────────────────────────────────────────────────╮
   │                                                               │
   │      New minor version of npm available! 6.4.1 -> 6.7.0       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v6.7.0   │
   │               Run npm install -g npm to update!               │
   │                                                               │
   ╰───────────────────────────────────────────────────────────────╯

 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile 
```

4.  js-templates
```
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile  make js-templates
/home/sujith/test/owncloud3/apps/impersonate/node_modules/handlebars/bin/handlebars -n "OCA.Impersonate" js/templates/impersonateNotification.handlebars -f js/templates/impersonateNotification.handlebars.js
/home/sujith/test/owncloud3/apps/impersonate/node_modules/handlebars/bin/handlebars -n "OCA.Impersonate" js/templates/addImpersonateIcon.handlebars -f js/templates/addImpersonateIcon.handlebars.js
/home/sujith/test/owncloud3/apps/impersonate/node_modules/handlebars/bin/handlebars -n "OCA.Impersonate" js/templates/removeImpersonateIcon.handlebars -f js/templates/removeImpersonateIcon.handlebars.js
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile 
```

5. build-src
```
 ✘ sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile  make build-src
mkdir -p build/impersonate
cp --parents -r \
        appinfo \
        controller \
        css \
        img \
        js \
        l10n \
        lib \
        README.md \
        LICENSE \
        CHANGELOG.md \
        templates \
        build/impersonate
rm -f build/impersonate/js/templates/*.handlebars
Skipping signing, either no key and certificate found in /home/sujith/.owncloud/certificates/impersonate.key and /home/sujith/.owncloud/certificates/impersonate.crt or occ can not be found at /home/sujith/test/owncloud3/apps/impersonate/../../occ
tar -czf build/impersonate.tar.gz -C /home/sujith/test/owncloud3/apps/impersonate/build/impersonate ../impersonate
tar: Removing leading `../' from member names
 sujith@sujith-ownCloud  ~/test/owncloud3/apps/impersonate   fix-makefile 
```

Let me know if this looks ok. @PVince81 @phil-davis @individual-it

sharidas commented:
#### Context ####
- This is a pure Makefile change. 
- Trying to add verbose messages in the make help, so that it would help users know how to use this app, before enabling it.

@phil-davis @individual-it Kindly help me review this change set.

phil-davis commented on README.md:
```diff
@@ -7,6 +7,8 @@ Once Impersonate is installed, a new column will be available in the user manage
 
 ## Installation
+For development, execute `make build-dep; make js-templates`
+To create distribution tar file, exceute `make dist`
```
s/exceute/execute

sharidas replied:
Done.

phil-davis commented:
It is still saying "This branch is out-of-date with the base branch". Needs rebase to latest master.

sharidas commented:
> It is still saying "This branch is out-of-date with the base branch". Needs rebase to latest master.

Ouch, just rebased :smile:  Thanks for pointint it out.
